.github/workflows/check.yml

```diff
@@ -71,3 +72,20 @@ jobs:
     - name: Build docs
       if: ${{matrix.build_docs}}
       run: cmake --build '${{github.workspace}}/build' --target doc
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: build results ${{ matrix.os }} ${{ matrix.build }}
+        path: ${{github.workspace}}/build
+        if-no-files-found: error
+
+    - name: Upload rdiff
+      uses: actions/upload-artifact@v2
+      if: ${{ ! matrix.no_rdiff }}
+      with:
+        name: rdiff ${{ matrix.os }} ${{ matrix.build }}
+        path: |
+            ${{github.workspace}}/build/rdiff
```

Comment:
Note that unless its a static build, the rdiff executable will also need the librsync library to run, which is either the .so (and related symlinks) on linux or the .dll on windows.

But perhaps, rather than manually selecting the bits to put in the artifact, perhaps we should do a `cmake --build '${{github.workspace}}/build' --target install` and then build an artifact out of everything that is installed. That way we should get the executables, supporting libraries, required headers to use those libraries, and supporting man pages.

That also means we don't need to selectively exclude the no-rdiff case, as we probably also want to build/check the artifact you get from installing only the library without rdiff.

I'm not sure what happens on windows when you run a 'cmake --target install`, and it wouldn't surprise me if it's broken. However, this is probably just something else we need to fix. 

The bad news is my experience with trying to make windows work for the github actions was kinda painful, but we are pretty close now to a working windows setup. At least github actions runs nice and fast for testing all this.

Reply:
I wonder if just building a self-contained static executable would be more helpful for people who want the executable?

Reply:
We now have install artifacts for all builds that we test. Currently, we only test static linking on Linux, so we don't have a static executable for windows, only a dynamic one. We could add another static build on windows, but I don't know if we need one.